base/show.jl

```diff
@@ -90,7 +91,7 @@ function show_default(io::IO, x::ANY)
     nf = nfields(t)
     if nf != 0 || t.size==0
         if !show_circular(io, x)
-            recur_io = IOContext(io, :SHOWN_SET => x)
+            recur_io = IOContext(io, SHOWN_SET=x, multiline=false)
```

Comment:
> That's a mouthful

OK, but "I'm not sure that we really want to always make 2x the number of IOContexts."

How does the performance of this form compare? This seems like it is the more performance-sensative path than static-show (https://github.com/JuliaLang/julia/pull/16049#issuecomment-214593490) and this makes the `O(n^2)` path (show_circular) more expensive.